velocity/src/main/java/com/github/games647/fastlogin/velocity/FastLoginVelocity.java

```diff
@@ -0,0 +1,192 @@
+ * SOFTWARE.
+ */
+package com.github.games647.fastlogin.velocity;
+
+import com.github.games647.fastlogin.core.AsyncScheduler;
+import com.github.games647.fastlogin.core.message.ChangePremiumMessage;
+import com.github.games647.fastlogin.core.message.ChannelMessage;
+import com.github.games647.fastlogin.core.message.SuccessMessage;
+import com.github.games647.fastlogin.core.shared.FastLoginCore;
+import com.github.games647.fastlogin.core.shared.PlatformPlugin;
+import com.github.games647.fastlogin.velocity.listener.ConnectListener;
+import com.github.games647.fastlogin.velocity.listener.PluginMessageListener;
+import com.google.common.collect.MapMaker;
+import com.google.common.io.ByteArrayDataOutput;
+import com.google.common.io.ByteStreams;
+import com.google.inject.Inject;
+import com.velocitypowered.api.command.CommandSource;
+import com.velocitypowered.api.event.Subscribe;
+import com.velocitypowered.api.event.proxy.ProxyInitializeEvent;
+import com.velocitypowered.api.plugin.Plugin;
+import com.velocitypowered.api.plugin.annotation.DataDirectory;
+import com.velocitypowered.api.proxy.Player;
+import com.velocitypowered.api.proxy.ProxyServer;
+import com.velocitypowered.api.proxy.messages.MinecraftChannelIdentifier;
+import com.velocitypowered.api.proxy.server.RegisteredServer;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import org.slf4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentMap;
+
+//TODO: Support for floodgate
+@Plugin(id = PomData.NAME, name = PomData.DISPLAY_NAME, description = PomData.DESCRIPTION, url = PomData.URL,
+        version = PomData.VERSION, authors = {"games647", "https://github.com/games647/FastLogin/graphs/contributors"})
+public class FastLoginVelocity implements PlatformPlugin<CommandSource> {
+
+    private final ProxyServer server;
+    private final Path dataDirectory;
+    private final Logger logger;
+    private final ConcurrentMap<InetSocketAddress, VelocityLoginSession> session = new MapMaker().weakKeys().makeMap();
+    private final String PROXY_ID_fILE = "proxyId.txt";
+
+    private FastLoginCore<Player, CommandSource, FastLoginVelocity> core;
+    private AsyncScheduler scheduler;
+    private UUID proxyId;
+
+    @Inject
+    public FastLoginVelocity(ProxyServer server, Logger logger, @DataDirectory Path dataDirectory) {
+        this.server = server;
+        this.logger = logger;
+        this.dataDirectory = dataDirectory;
+    }
+
+    @Subscribe
+    public void onProxyInitialization(ProxyInitializeEvent event) {
+        scheduler = new AsyncScheduler(logger, getThreadFactory());
+        core = new FastLoginCore<>(this);
```

Comment:
Please add a close call for the core component. It will make sure that we are safely closing our resources, mainly the Hikari pool and our own scheduler that runs tasks velocity is not aware of.

Reply:
Example:

https://github.com/games647/FastLogin/blob/ec7c421f839df286ceae52d1fb89d48556972e3a/bungee/src/main/java/com/github/games647/fastlogin/bungee/FastLoginBungee.java#L103